diff --git a/ext/Kbuild b/ext/Kbuild
index 600d758..e8589a3 100644
--- a/ext/Kbuild
+++ b/ext/Kbuild
@@ -3,3 +3,4 @@ obj-y += lib/
 obj-y += fs/
 obj-y += debug/
 obj-y += lib/mraa/
+obj-y += lib/upm/
diff --git a/ext/Kconfig b/ext/Kconfig
index fcc125d..8ad430a 100644
--- a/ext/Kconfig
+++ b/ext/Kconfig
@@ -18,4 +18,6 @@ source "ext/debug/Kconfig"
 
 source "ext/lib/mraa/Kconfig"
 
+source "ext/lib/upm/Kconfig"
+
 endmenu
diff --git a/ext/Makefile b/ext/Makefile
index 087d210..a379914 100644
--- a/ext/Makefile
+++ b/ext/Makefile
@@ -6,3 +6,4 @@ ifdef CONFIG_MRAA
     ZEPHYRINCLUDE += -I$(srctree)/ext/lib/mraa/include
 endif
 include $(srctree)/ext/lib/mraa/Makefile
+include $(srctree)/ext/lib/upm/Makefile
