diff --git a/ext/Kbuild b/ext/Kbuild
index 97cb478..0e8e091 100644
--- a/ext/Kbuild
+++ b/ext/Kbuild
@@ -1,3 +1,5 @@
 obj-y += hal/
 obj-y += lib/
 obj-y += fs/
+obj-$(CONFIG_MRAA) += lib/mraa/
+obj-y += lib/upm/
diff --git a/ext/Kconfig b/ext/Kconfig
index 9d26499..a61c4ee 100644
--- a/ext/Kconfig
+++ b/ext/Kconfig
@@ -24,4 +24,8 @@ source "ext/lib/crypto/Kconfig"
 
 source "ext/fs/Kconfig"
 
+source "ext/lib/mraa/Kconfig"
+
+source "ext/lib/upm/Kconfig"
+
 endmenu
diff --git a/ext/Makefile b/ext/Makefile
index 45b3952..7d4a5c2 100644
--- a/ext/Makefile
+++ b/ext/Makefile
@@ -1,3 +1,7 @@
 include $(srctree)/ext/lib/Makefile
 include $(srctree)/ext/hal/Makefile
 include $(srctree)/ext/fs/Makefile
+ifdef CONFIG_MRAA
+ZEPHYRINCLUDE += -I$(srctree)/ext/lib/mraa/include
+endif
+include $(srctree)/ext/lib/upm/Makefile
